### PR TITLE
Cacert rationalization

### PR DESCRIPTION
**Pull request by thetrime**
This isn't quite ready for merge just yet - I want to fix the last problem outlined in #148

### CODE REVIEW

File: ssl.pl [hunk 1]
```diff
@@ -312,7 +312,30 @@
 
 ssl_context(Role, SSL, Module:Options) :-
     select_option(ssl_method(Method), Options, O1, sslv23),
-    '_ssl_context'(Role, SSL, Module:O1, Method).
+    upgrade_legacy_options(O1, O2),
+    % If the options list does not contain a cacerts/1 term (including cacerts([]))
+    % then assume that the system root certificates should be used
+    (  memberchk(cacerts(_), O2)
```
Comment (JanWielemaker) [inline ssl.pl]: This should be option/2, no?

File: ssl.pl [hunk 2]
```diff
@@ -312,7 +312,30 @@
...
+    -> O3 = O2
+    ;  O3 = [cacerts([system(root_certificates)])|O2]
+    ),
+    '_ssl_context'(Role, SSL, Module:O3, Method).
+
+upgrade_legacy_options(O1, O4):-
+    % If the options list contains cacert_file/1 then update the cacerts list
+    select(cacert_file(CACertFile), O1, O2),
```
Comment (JanWielemaker) [inline ssl.pl]: And this select_option/3, no?

Comment (thetrime) [reply ssl.pl]: Ok. There are some other things to take care of as well. Thanks for the feedback!

File: ssl.pl (hunk 1 above)
Comment (thetrime) [reply ssl.pl]: If you prefer option(cacerts(_), O2, _) to memberchk(cacerts(_), O2) then that's fine by me. The point is that we just want to check if an cacerts/1 term is in there - and only do something if it is *not*

Comment (JanWielemaker) [reply ssl.pl]: I know.  option/2 is more explicit we are processing options and allows for the alternative dict based option format and/or other future changes.

File: ssl.pl (hunk 2 above)
Comment (thetrime) [reply ssl.pl]: Is there a standard way to print such a warning? A grep of lines containing both 'warning' and 'deprecated' on the source doesn't return much

Comment (JanWielemaker) [reply ssl.pl]: Using `print_message(warning, deprecated(Term))`.    The '$messages':deprecated//1 should become hookable as well (as prolog:deprecated//1).

**Comment by JanWielemaker** [comment]
Do you signal me if this should be merged?

**Comment by thetrime** [comment]
Sure. I'm just waiting for your reply to

> Is there a standard way to print such a warning? A grep of lines containing both 'warning' and 'deprecated' on the source doesn't return much

**Comment by JanWielemaker** [comment]
> Sure. I'm just waiting for your reply to

?  I answered to that, but in the review action.  Anyway, see boot/messages.pl from wich we deduce

    print_message(warning, deprecated(Term)).

Only, the deprecated//1 there should be hooked as prolog:deprecated//1 as all the others.  Please add or leave to me.

**Comment by thetrime** [comment]
OK. I'll leave the translation into a message to you. My functional testing is finished now - my tests all pass, and I can't generate any crashes - but I haven't done any checks with valgrind for leaks. I'm still waiting on my Ubuntu VM to upgrade so I can install valgrind...

**Comment by JanWielemaker** [comment]
Ok.  Squash-merged after some more minor changes.  Also involved the `http` package and the core system for the deprecated messages.   It doesn't deserve the beauty contest, but it is cleaner now and the transition should have marginal impact.

**Pull request closed without merging**